### PR TITLE
Code Quality: Introduce details layout columns generator phase 2

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Summary

This is the phase 2 of creating a generator of columns for DetailsLayout page.

We expect that this will significantly reduce the burden on developers implementing new columns to actually make additional changes, significantly reduce the amount of XAML code in DetailsLayoutPage, keeping the codebase clean, and allowing us to easily refactor.

## Task Checklist

- Delete `ColumnsViewModel`
- Rename `FolderSettings` properties with `LayoutPreferencesManager`
- Added a new factory for the generator of columns
- Added a new setting named `DefaultColumnItems` collection in the app json settings
- Added a new converter named `DoubleToGridLength` to avoid relying on UI class in the model/item class.
- Added ViewModel for DetailsLayoutPage
- Moved columns-related methods/commands to the ViewModel

Note that there's a small amount of refactoring on the codebase.

- [x] Sort indicator behavior
- [x] Resizing behavior
- [ ] Show/Hide column toggle behavior
- [ ] Git repository folder
- [ ] Recycle Bin folder

## Steps To Validate Changes

1. Launch the app
2. Go to Desktop
3. Resize random column
4. Double click a resizer
5. Click a column to sort
6. Hide a column from the menu flyout
7. Go to Downloads
8. Go back to Desktop
9. See all settings you set is applied correctly
10. Go to a random folder with DetailsLayout
11. Try 3-9 randomly

## Known Issues

_N/A_

## PR Checklist

- [x] **Development**: #12505
- [x] **Approval**: Have discussed with and got approval from the team[^1]
- [x] **Tests**: Tested accessibility on the local end
- [x] **Deployment**: Deployed the app on the local end
- [ ] **Localization**: Modified en-US string resources[^2]
- [ ] **Co-authors**: _N/A_

## Screenshots

_N/A_

[^1]: If the request hasn't been agreed by the team, this work might be rejected. Make sure that you get approval from the team before opening any PR related the request.
[^2]: If you removed any en-US string resources, make sure that there are no references of those resources. When you add a new en-US string resources, do not make any changes on other languages' resources; [Crowdin](https://crowdin.com/project/files-app) will do that, instead.

### CODE REVIEW

**Comment by 0x5bfa** [comment]
I have a last issue where the columns are messed up when I show/hide a column in context menu flyout. Otherwise, works really good.

File: src/Files.App/Data/Items/DetailsLayoutColumnItem.cs
Comment (yaira2) [inline src/Files.App/Data/Items/DetailsLayoutColumnItem.cs:1]: Will these changes reset existing settings?

**Review by yaira2 — COMMENTED** [review]
Code changes look good but I'll need a little more time to test the behavior

**Comment by yaira2** [comment]
The git flyout appears to be empty
image

File: src/Files.App/Data/Items/DetailsLayoutColumnItem.cs
Comment (0x5bfa) [reply src/Files.App/Data/Items/DetailsLayoutColumnItem.cs:1]: Looks in that way. I'm gonna look into it.

**Comment by 0x5bfa** [comment]
Ugh... I could not figure out why the initialization of column items is executed after fetching the columns (that was actually why the columns width messed up when navigating away from another folder).

**Pull request closed without merging**